### ISSUE TITLE
Support for emoji hashtags

### ISSUE THREAD

**Issue by akhilacubrio** [Hypercontext/linkifyjs#419]
Occurs on social media. E.g.: #🐶

**Comment by nfrasser** [comment]
Hi @akhilacubrio, thanks for the suggestions, I will consider it for a future release. As a workaround, you can enable this with a custom plugin like this:

```js
import { State, createTokenClass, registerPlugin } from "linkifyjs";

const HashtagToken = createTokenClass('hashtag', { isLink: true });

registerPlugin('hashtag', ({ scanner, parser }) => {
  // Various tokens that may compose a hashtag
  const { POUND, UNDERSCORE } = scanner.tokens;
  const { alpha, numeric, alphanumeric, emoji } = scanner.tokens.groups;

  // Take or create a transition from start to the '#' sign (non-accepting)
  // Take transition from '#' to any text token to yield valid hashtag state
  // Account for leading underscore (non-accepting unless followed by alpha)
  const Hash = parser.start.tt(POUND);
  const HashPrefix = Hash.tt(UNDERSCORE);
  const Hashtag = new State(HashtagToken);

  Hash.ta(numeric, HashPrefix);
  Hash.ta(alpha, Hashtag);
  Hash.ta(emoji, Hashtag);
  HashPrefix.ta(alpha, Hashtag);
  HashPrefix.ta(emoji, Hashtag);
  HashPrefix.ta(numeric, HashPrefix);
  HashPrefix.tt(UNDERSCORE, HashPrefix);
  Hashtag.ta(alphanumeric, Hashtag);
  Hashtag.ta(emoji, Hashtag);
  Hashtag.tt(UNDERSCORE, Hashtag); // Trailing underscore is okay
});
```

**Comment by nfrasser** [comment]
Fixed in v4.1.0
